audio/pwaudio.c

```diff
@@ -22,7 +26,18 @@ typedef struct
 
     struct pw_core *core;
     struct spa_hook core_listener;
+    struct pw_stream *stream;
+	struct spa_hook stream_listener;
     struct spa_source *timer;
+    
+    uint32_t frame_size;
+    struct spa_ringbuffer ring;
+    void *buffer;
+
+    #define MODE_SINK	0
```

Comment:
I would put these `defines` in the global space. Actually, probably you want an `enum` instead?

Reply:
These two defines are only relevant to the PWVoice struct, why should it be defined globally?

Reply:
Maybe I am C++ biased, but it looks weird that defines are not grouped closer together (usually in the global space), so that they are easier to find. I understand that in this case defines are closer to the variable that will take those values, but this can be achieved with an `enum`, with a few advantages imo:
- You do not need to explicitly set each value of the set (the compiler does the numbering for you). So you can add new states in between without having to reset any numbering.
- Compiler warns you for unused values in `switch` statements.
- The type is defined so that the variable in the struct does not take invalid (non-defined) values.

In the long term will make it more maintainable, but if you do not expect it to grow it can stay as is,

Reply:
fair enough, I don't object to using enums.